src/Service.Tests/GraphQLBuilder/NestedMutationBuilderTests.cs

```diff
@@ -0,0 +1,408 @@
+        /// section in the configuration of the source entity.</param>
+        [DataTestMethod]
+        [DataRow("Book", "Author", DisplayName = "Validate creation of source->target linking input object for Book->Author M:N relationship")]
+        [DataRow("Author", "Book", DisplayName = "Validate creation of source->target linking input object for Author->Book M:N relationship")]
+        [DataRow("BookNF", "AuthorNF", DisplayName = "Validate creation of source->target linking input object for BookNF->AuthorNF M:N relationship")]
+        [DataRow("AuthorNF", "BookNF", DisplayName = "Validate creation of source->target linking input object for AuthorNF->BookNF M:N relationship")]
+        public void ValidateCreationOfSourceTargetLinkingInputForMNRelationship(string sourceEntityName, string targetEntityName)
+        {
+            NameNode inputTypeNameForBook = CreateMutationBuilder.GenerateInputTypeName(GetDefinedSingularName(
+                sourceEntityName,
+                _runtimeConfig.Entities[sourceEntityName]));
+            Entity entity = _runtimeConfig.Entities[sourceEntityName];
+            InputObjectTypeDefinitionNode inputObjectTypeDefinition = (InputObjectTypeDefinitionNode)_mutationDefinitions.FirstOrDefault(d => d.Name.Value.Equals(inputTypeNameForBook.Value));
+            NameNode inputTypeName = CreateMutationBuilder.GenerateInputTypeName(GenerateLinkingNodeName(
+                GetDefinedSingularName(sourceEntityName, _runtimeConfig.Entities[sourceEntityName]),
+                GetDefinedSingularName(targetEntityName, _runtimeConfig.Entities[targetEntityName])));
+            List<InputValueDefinitionNode> inputFields = inputObjectTypeDefinition.Fields.ToList();
+            int indexOfRelationshipField = inputFields.FindIndex(field => field.Type.InnerType().NamedType().Name.Value.Equals(inputTypeName.Value));
+            Assert.AreNotEqual(-1, indexOfRelationshipField);
+        }
+
+        /// <summary>
+        /// Test to validate that the linking input types generated for a source->target relationship contains input fields for:
+        /// 1. All the fields belonging to the target entity, and
+        /// 2. All the non-relationship fields in the linking entity.
+        /// </summary>
+        /// <param name="sourceEntityName">Name of the source entity for which the configuration is provided in the config.</param>
+        /// <param name="targetEntityName">Name of the target entity which is related to the source entity via a relationship defined in the 'relationships'
+        /// section in the configuration of the source entity.</param>
+        [DataTestMethod]
+        [DataRow("Book", "Author",
+            DisplayName = "Validate presence of target fields and linking object fields in source->target linking input object for Book->Author M:N relationship")]
+        [DataRow("Author", "Book",
```

Comment:
What if a relationship is exposed only in one direction in the configuration file?

Reply:
The relationship between `stocks` and `stocks_price` is exposed only in the `stocks_price` table.